Dockerfile

```diff
@@ -0,0 +1,41 @@
+ARG RUBY_VERSION=2.7.1
+FROM ruby:$RUBY_VERSION-alpine as hyrax-base
+
+ARG DATABASE_APK_PACKAGE="postgresql-dev"
+ARG EXTRA_APK_PACKAGES="git"
+
+RUN apk --no-cache upgrade && \
+  apk --no-cache add build-base \
+  tzdata \
+  nodejs \
+  $DATABASE_APK_PACKAGE \
+  $EXTRA_APK_PACKAGES
+
+RUN gem update bundler
+
+RUN mkdir -p /app-data/samvera/hyrax-webapp
+WORKDIR /app-data/samvera/hyrax-webapp
+
+COPY ./bin /app-data/samvera
+
+CMD ["bundle", "exec", "puma", "-v", "-b", "tcp://0.0.0.0:3000"]
```

Comment:
thanks! i'm not sure why i didn't consider that an entrypoint script could solve the permissions issues.

it seems like a nice thing to have in place anyway.

Reply:
i pushed up a refined version of this, but i'm still running into a major permissions issue. since the `app` user doesn't have permissions on `tmp` the `chown` commands fail.

my current work around is to use `user: root` within `docker-compose`. this is vaguely okay, since these files will have been `chown`d on `COPY` during the build. it's pretty fragile, though, since this is a pretty big difference between the image defaults and the `docker-compose` setup.

thoughts?

Reply:
People solve this a bunch of different ways "in the wild"
- in the passenger images they run as root, but su-exec (or gosu, not sure which) the process to be app. That means things like permissions changes and shelling in happen as root, which to be honest is very convenient and saves a lot of working around things. 

- in the bitnami wordpress image they run the whole thing at app, and you get stuck trying to log in to the node in order to get anything in root permissions shell wise in K8S, which is a pain.

Reply:
it would be really nice if docker just let do `chown` or set mode on volume mount...

i'm inclined to move this through running as root from compose, and see what happens from there.